pkg/controller/reconciler/deployment_client.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	aztoken "github.com/radius-project/radius/pkg/azure/tokencredentials"
+	"github.com/radius-project/radius/pkg/sdk"
+	sdkclients "github.com/radius-project/radius/pkg/sdk/clients"
+)
+
+// DeploymentClient is an interface for interacting with
+// UCP ResourceDeploymentsClient.
+type DeploymentClient interface {
+	ResourceDeployments() ResourceDeploymentsClient
+}
+
+// ResourceDeploymentsClient is an interface for interacting
+// with UCP Deployments.
+type ResourceDeploymentsClient interface {
```

Comment:
I'm a little confused by the naming choice. We already have a `ResourceDeploymentsClient` used in the SDK clients package: https://github.com/radius-project/radius/blob/72de369155eeb9bcbcf79e9ca68a2cc994e34749/pkg/controller/reconciler/deployment_client.go#L32

Based on the code, it seems like these are both deployment clients that serve UCP (although the one linked above serves Azure and UCP) so why are we redefining the interface? If both `ResourceDeploymentsClient are meant to serve different purposes or serve different deployments (Azure/UCP vs deployment resources/templates), I think we should rename this for clarity.

Reply:
They do the same thing - I think we could consolidate them. I think I would rather handle this together with other changes later to move the mocks outside of `reconciler` package though.